code/edgeidentity/src/test/java/com/adobe/marketing/mobile/edge/identity/IdentityStateTests.java

```diff
@@ -542,4 +561,223 @@ public void testUpdateLegacyExperienceCloudId_notSetWhenExistingIsNull() {
+			MobileCore.dispatchEvent(consentEventCaptor.capture(), any(ExtensionErrorCallback.class));
+
+			Event consentEvent = consentEventCaptor.getAllValues().get(0);
+
+			Map<String, String> consentEventData = flattenMap(consentEvent.getEventData());
+			// `flattenMap` allows for checking the keys' hierarchy and literal values simultaneously
+			assertEquals("GAID", consentEventData.get("consents.adID.idType"));
+			assertEquals(expectedConsent, consentEventData.get("consents.adID.val"));
+		}
+
+		if (isSharedStateUpdateExpected) {
+			// Verify persistent store
+			final ArgumentCaptor<String> persistenceValueCaptor = ArgumentCaptor.forClass(String.class);
+			verify(mockSharedPreferenceEditor, times(1))
+				.putString(eq(IdentityConstants.DataStoreKey.IDENTITY_PROPERTIES), persistenceValueCaptor.capture());
+			Map<String, String> persistedData = flattenJSONString(persistenceValueCaptor.getAllValues().get(0));
+			verifyFlatIdentityMap(expectedAdId, state.getIdentityProperties().getECID().toString(), persistedData);
+
+			// Verify shared state and properties
+			assertEquals(1, setXDMSharedEventStateCalledTimes);
+		} else {
+			// Verify persistent store
+			verify(mockSharedPreferenceEditor, never())
+				.putString(eq(IdentityConstants.DataStoreKey.IDENTITY_PROPERTIES), any(String.class));
+
+			// Verify shared state and properties
+			assertEquals(0, setXDMSharedEventStateCalledTimes);
+		}
+		// Verify identity map
+		final Map<String, String> flatIdentityMap = flattenMap(state.getIdentityProperties().toXDMData(false));
+		verifyFlatIdentityMap(expectedAdId, state.getIdentityProperties().getECID().toString(), flatIdentityMap);
+		// Verify shared state and properties
+		assertEquals(expectedAdId, state.getIdentityProperties().getAdId());
+	}
+
+	// Test helpers
+
+	/**
+	 * Creates an event with the given adId in data
+	 * @param adId
+	 * @return
+	 */
+	private Event fakeGenericIdentityEvent(final String adId) {
+		return new Event.Builder(
+			"Test event",
+			IdentityConstants.EventType.GENERIC_IDENTITY,
+			IdentityConstants.EventSource.REQUEST_IDENTITY
+		)
+			.setEventData(
+				new HashMap<String, Object>() {
+					{
+						put(IdentityConstants.EventDataKeys.ADVERTISING_IDENTIFIER, adId);
+					}
+				}
+			)
+			.build();
+	}
+
+	/**
+	 * Verifies the flat map contains the required ad ID and ECID
+	 * Valid ECID string and flat identity map is always required
+	 * @param adId the ad ID to check, can be null if no ad ID should be present; then the absence of ad ID will be verified
+	 * @param ecid the ECID string to check; must not be null (since this is the booted state)
+	 * @param flatIdentityMap the flat identity map to check
+	 * @return true if identity map contains the required identity properties, false otherwise
+	 */
+	private void verifyFlatIdentityMap(
+		@Nullable final String adId,
+		@NonNull final String ecid,
+		@NonNull final Map<String, String> flatIdentityMap
```

Comment:
let's reverse the order of the parameters here, similar with the other helper and the junit functions use actual values first, then expected ones. reading:
verifyFlatIdentityMap(flatIdentityMap, expectedAdId, expectedECID)